src/apps/invitations/services.py

```diff
@@ -70,31 +71,31 @@ async def send_respondent_invitation(
         self,
         applet_id: uuid.UUID,
         schema: InvitationRespondentRequest,
-        subject_id: uuid.UUID,
+        subject: Subject,
     ) -> InvitationDetailForRespondent:
         await self._is_validated_role_for_invitation(applet_id, Role.RESPONDENT)
 
-        subject_invitation = await self.invitations_crud.get_pending_subject_invitation(applet_id, subject_id)
+        subject_invitation = await self.invitations_crud.get_pending_subject_invitation(applet_id, subject.id)
         if subject_invitation and subject_invitation.email != schema.email:
             raise ValidationError("Subject already invited with different email.")
 
         # Get invited user if he exists. User will be linked with invitaion
         # by user_id in this case
         invited_user = await UsersCRUD(self.session).get_user_or_none_by_email(email=schema.email)
         invited_user_id = invited_user.id if invited_user is not None else None
-        meta = RespondentMeta(subject_id=str(subject_id))
+        meta = RespondentMeta(subject_id=str(subject.id))
         payload = {
-            "email": schema.email,  # TODO remove?
+            "email": schema.email,
             "applet_id": applet_id,
             "role": Role.RESPONDENT,
             "key": uuid.uuid3(uuid.uuid4(), schema.email),
             "invitor_id": self._user.id,
             "status": InvitationStatus.PENDING,
-            "first_name": schema.first_name,  # TODO remove
-            "last_name": schema.last_name,  # TODO remove
+            "first_name": subject.first_name,
```

Comment:
For respondent invittation first_name, last_name, nickname should be stored in subject record, not in invitation